src/Integration.Vsix/Suppression/DelegateInjector.cs

```diff
@@ -0,0 +1,92 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+ using System;
+using System.Reflection;
+using Microsoft.CodeAnalysis;
+
+/* We want to suppress issues from Sonar analyzers that are added via NuGet.
+ * If the analyzer in the NuGet is of a different version from the one in the VSIX then just
+ * setting the static property on the version referenced by the VSIX won't work (both analyzer
+ * assemblies will be loaded in memory, each with its own static class + property).
+ * We need to set the static property for each Sonar analyzer assembly that is loaded.
+ * 
+ * Version-compatibility: the NuGet package and VSIX might reference different versions of Roslyn etc,
+ * but we need to be able to assign a delegate from the VSIX to the NuGet static property.
+ * If the NuGet package is using a higher version of Roslyn then the analyzer won't work anyway.
+ * If is is using a lower version of Roslyn then the binding redirect in VS should make the assignment works.
+ */
+
+namespace SonarLint.VisualStudio.Integration.Vsix.Suppression
+{
+    /// <summary>
+    /// Injects the suppression calculator function into any versions of the Sonar
+    /// analyzers that are already loaded, or in to any new versions as they are loaded
+    /// </summary>
+    internal class DelegateInjector : IDisposable
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Fix this implementation of 'IDisposable' to conform to the dispose pattern. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS3881)